checks.sh

```diff
@@ -1,48 +1,110 @@
 #! /bin/sh
+# arguments:
+# 1. a relative path to a directory
+# 2. flag to flip exit codes for success and failure (used in testing)
+# 3. flag to ignore whitespace when diffing
 
-# check if CITATION.cff exists
-if [ -f "CITATION.cff" ]; then
-    echo "(1/7) CITATION.cff exists" ;
+if [ -z "$2" ] || [ "$2" == "0" ] ; then
+    FAILURE_EXPECTED=0
+    FAILURE_CODE=1
+    SUCCESS_CODE=0
+elif [ "$2" == "1" ] ; then
+    echo "Reversing the definition of failure and success"
+    FAILURE_EXPECTED=1
+    FAILURE_CODE=0
+    SUCCESS_CODE=1
 else
-    echo "(1/7) CITATION.cff missing. You can use https://bit.ly/cffinit to create one.";
+    echo "Second input argument should be empty, 0, or 1. Aborting."
     exit 1;
 fi
 
+# If the user has provided a first input argument, interpret
+# it as a relative path to a directory, and change into it
+if [ -n "$1" ] ; then
+    cd $1 || exit ${FAILURE_CODE}
+    echo "Changed directory into $1"
+fi
 
-# check if .zenodo.json exists
-if [ -f ".zenodo.json" ]; then
-    echo "(2/7) .zenodo.json exists" ;
+if [ "$3" == "0" ] ; then
+    echo "Not ignoring the whitespace when diff'ing"
+    DIFF_IGNORE_WHITESPACE=0
+elif [ -z "$3" ] || [ "$3" == "1" ] ; then
+    DIFF_IGNORE_WHITESPACE=1
 else
-    echo "(2/7) .zenodo.json missing. Aborting.";
+    echo "Third input argument should be empty, 0, or 1. Aborting."
```

Comment:
What is the third argument? Please name it.

Reply:
added